### PR TITLE
Prefer constructor injection in documentation

### PR DESCRIPTION
**Pull request by gmethvin**
As discussed in #5837, constructor injection is easier to understand for newbies to DI, and is the right way to use DI in a real application, so I've converted all the documentation examples to use constructor injection instead of field injection.

### CODE REVIEW

File: documentation/manual/working/commonGuide/filters/code/detailedtopics/configuration/cors/Filters.java [hunk 1]
```diff
@@ -12,11 +12,17 @@
 
 public class Filters implements HttpFilters {
 
+    private EssentialFilter[] filters;
+
     @Inject
-    CORSFilter corsFilter;
+    public Filters(CORSFilter corsFilter) {
+        this.filters = new EssentialFilter[] {
```
Comment (marcospereira) [inline documentation/manual/working/commonGuide/filters/code/detailedtopics/configuration/cors/Filters.java]: Is this against [guice best practices](https://github.com/google/guice/wiki/InjectOnlyDirectDependencies)?

File: documentation/manual/working/javaGuide/main/sql/code/JavaApplicationDatabase.java [hunk 2]
```diff
@@ -9,6 +9,12 @@
 import play.db.*;
 
 class JavaApplicationDatabase extends Controller {
-    @Inject Database db;
+
+    private Database db;
+
+    public JavaApplicationDatabase(Database db) {
```
Comment (marcospereira) [inline documentation/manual/working/javaGuide/main/sql/code/JavaApplicationDatabase.java]: Missing `@Inject` annotation?

**Comment by marcospereira** [comment]
@gmethvin this now have conflicts because of #6117.

File: documentation/manual/working/javaGuide/main/sql/code/JavaApplicationDatabase.java (hunk 2 above)
Comment (gmethvin) [inline documentation/manual/working/javaGuide/main/sql/code/JavaApplicationDatabase.java]: Thanks. Updated.

File: documentation/manual/working/commonGuide/filters/code/detailedtopics/configuration/cors/Filters.java (hunk 1 above)
Comment (gmethvin) [inline documentation/manual/working/commonGuide/filters/code/detailedtopics/configuration/cors/Filters.java]: Not sure what you mean. `CORSFilter` is still a direct dependency of `Filters`. We're just constructing the filters up front here.

**Comment by gmethvin** [comment]
I'm going to say this should also go into 2.5.x (possibly 2.4.x) since a lot of people are migrating to DI in this version.

**Pull request merged**